pkg/api/mlflow/service/run/service.go

```diff
@@ -444,10 +448,13 @@ func (s Service) SetRunTag(ctx context.Context, req *request.SetRunTagRequest) e
 
 	run, err := s.runRepository.GetByID(ctx, req.RunID)
 	if err != nil {
-		return api.NewResourceDoesNotExistError("unable to find run '%s': %s", req.RunID, err)
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			return api.NewResourceDoesNotExistError("Unable to find active run '%s'", req.RunID)
+		}
+		return api.NewResourceDoesNotExistError("Unable to find active run '%s': %s", req.RunID, err)
 	}
 	if !run.IsLifecycleStageActive() {
-		return api.NewResourceDoesNotExistError("unable to find active run '%s'", req.RunID)
+		return api.NewResourceDoesNotExistError("Unable to find active run '%s'", req.RunID)
```

Comment:
Let's change these messages everywhere to:
- `"Run '%s' not found"` when `err != nil`
- `"The run %s must be in 'active' lifecycle_stage"` when `!run.IsLifecycleStageActive()`

(these are the expected error messages in MLFlow and would allow us to patch their integration tests less)

Reply:
ook.

Reply:
@jgiannuzzi Python integration tests are waiting exactly `Unable to find active run` in case of error.

Reply:
I did some changes. waiting for tests.

Reply:
ook, now good. fixed it a bit in different way, but right now this is exactly what Python test expects.

Reply:
so of course the patch for the Python integration tests should be changed!
we can drop these parts of the patch essentially:
* https://github.com/G-Research/fasttrackml/blob/d2aa409d1581702de4e6516cf73f2934c2131c8b/tests/integration/python/mlflow/custom.patch#L166-L167
* https://github.com/G-Research/fasttrackml/blob/d2aa409d1581702de4e6516cf73f2934c2131c8b/tests/integration/python/mlflow/custom.patch#L173-L174

Reply:
this is really no big deal and I'm happy to have this in a separate PR later down the line — just create an issue for it so we don't forget about it

Reply:
ahhh, this is our patch. Can we talk about that today?